static/js/containers/pages/CheckoutPage.js

```diff
@@ -64,6 +69,30 @@ export class CheckoutPage extends React.Component<Props, State> {
     errors:       null
   }
 
+  componentDidMount = async () => {
+    const {
+      fetchBasket,
+      location: { search }
+    } = this.props
+    const params = queryString.parse(search)
+    const productId = parseInt(params.product)
+    if (!productId) {
+      await fetchBasket()
+      return
+    }
+
+    try {
+      await this.updateBasket({ items: [{ id: productId }] })
+
+      const couponCode = params.code
+      if (couponCode) {
+        await this.updateBasket({ coupons: [{ code: couponCode }] })
```

Comment:
This is a little tricky because the behavior we want is to update `items` then update `coupons`. If the `coupons` update fails the `items` update should still succeed. If both were sent as the same PATCH request and the `coupons` had a validation error but the `items` succeeded, I think the proper thing is to not make any changes and return a 400 error with an error message in the payload. What do you think?